SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+	bool FCheckActorsHaveSameAuth::Update()
+	{
+		VirtualWorkerId FirstActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[FirstActorName]);
+		VirtualWorkerId SecondActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[SecondActorName]);
+
+		Test->TestEqual(
+			FString::Printf(TEXT("Checking actors have the same auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckActorsHaveDifferentAuth, FAutomationTestBase*, Test, FName, FirstActorName, FName, SecondActorName);
+	bool FCheckActorsHaveDifferentAuth::Update()
+	{
+		VirtualWorkerId FirstActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[FirstActorName]);
+		VirtualWorkerId SecondActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[SecondActorName]);
+
+		Test->TestNotEqual(
+			FString::Printf(TEXT("Checking actors have different auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FCheckRequiresHandover, FAutomationTestBase*, TEST, bool, Expected);
+	bool FCheckRequiresHandover::Update()
+	{
+		bool Actual = Strat->RequiresHandoverData();
+		TEST->TestEqual(
+			FString::Printf(TEXT("Checking strategy requires handover data Expected: %b Actual: %b"), Expected, Actual),
+			Expected, Actual);
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FCheckShouldHaveAuthMatchesWhoShouldHaveAuth, FAutomationTestBase*, Test, FName, ActorName);
+	bool FCheckShouldHaveAuthMatchesWhoShouldHaveAuth::Update()
+	{
+		bool WeShouldHaveAuthority
+			= Strat->WhoShouldHaveAuthority(*TestActors[ActorName]) == Strat->GetLocalVirtualWorkerId();
+
+		Test->TestEqual(
+			FString::Printf(TEXT("Expected: %b Actual: %b"), WeShouldHaveAuthority, Strat->ShouldHaveAuthority(*TestActors[ActorName])),
+			WeShouldHaveAuthority, Strat->ShouldHaveAuthority(*TestActors[ActorName]));
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(GIVEN_strat_is_not_ready_WHEN_local_virtual_worker_id_is_set_THEN_is_ready)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy({}, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckStratIsReady(this, false));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckStratIsReady(this, true));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(GIVEN_layered_strat_of_three_two_by_four_grid_strats_and_default_strat_WHEN_get_minimum_required_workers_called_THEN_twenty_five_returned)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+
+		std::vector<StrategyConfig> ComponentStrategyConfig{ 3, {UTwoByFourLBGridStrategy::StaticClass(), {}} };
```

Comment:
If you refactor SetupStrategy as above, it would be good to have different sized layers for this test.  (Maybe just the TwoByFour and a singleton) Just to make sure that the iteration over the layers is working, and not just calling GetMinimum on the first layer 3 times.

Reply:
done